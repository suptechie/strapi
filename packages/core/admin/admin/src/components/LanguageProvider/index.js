/*
 *
 * LanguageProvider
 *
 * this component connects the redux state language locale to the
 * IntlProvider component and i18n messages (loaded from `app/translations`)
 */

import React, { useReducer } from 'react';
import PropTypes from 'prop-types';
import { IntlProvider } from 'react-intl';
import defaultsDeep from 'lodash/defaultsDeep';
import LocalesProvider from '../LocalesProvider';
import init from './init';
import reducer, { initialState } from './reducer';

const LanguageProvider = ({ children, localesNativeNames, messages }) => {
  const [{ locale }, dispatch] = useReducer(reducer, initialState, () => init(localesNativeNames));

  const changeLocale = locale => {
    dispatch({
      type: 'CHANGE_LOCALE',
      locale,
    });
  };

  const appMessages = defaultsDeep(messages[locale], messages.en);

  return (
    <IntlProvider locale={locale} defaultLocale="en" messages={appMessages} textComponent="span">
      <LocalesProvider
        changeLocale={changeLocale}
        localesNativeNames={localesNativeNames}
        messages={appMessages}
      >
        {React.Children.only(children)}
      </LocalesProvider>
    </IntlProvider>
  );
};

LanguageProvider.propTypes = {
  children: PropTypes.element.isRequired,
  localesNativeNames: PropTypes.object.isRequired,
  messages: PropTypes.object.isRequired,
};

export default LanguageProvider;
