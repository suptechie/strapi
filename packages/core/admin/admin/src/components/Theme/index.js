import React from 'react';
import { ThemeProvider } from '@strapi/parts/ThemeProvider';
import { ThemeProvider as TP } from 'styled-components';
import PropTypes from 'prop-types';
import { lightTheme } from '@strapi/parts';
import themes from '../../themes';
import GlobalStyle from '../GlobalStyle';

const Theme = ({ children, theme }) => (
  <ThemeProvider theme={theme}>
    <TP theme={themes}>
      {children}
      <GlobalStyle />
    </TP>
  </ThemeProvider>
);

Theme.propTypes = {
  children: PropTypes.element.isRequired,
  theme: PropTypes.object,
};

Theme.defaultProps = {
  theme: lightTheme,
};

export default Theme;
