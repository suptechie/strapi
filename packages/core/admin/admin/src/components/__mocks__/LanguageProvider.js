export function useLocalesProvider() {
  return {
    changeLocale() {},
    localeNames: { en: 'English' },
    messages: ['test'],
  };
}
