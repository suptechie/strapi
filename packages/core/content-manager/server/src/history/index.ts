import type { Plugin } from '@strapi/types';
import { controllers } from './controllers';
import { services } from './services';
import { routes } from './routes';
import { getService } from './utils';
import { historyVersion } from './models/history-version';

/**
 * Check once if the feature is enabled (both license info & feature flag) before loading it,
 * so that we can assume it is enabled in the other files.
 */
const getFeature = (): Partial<Plugin.LoadedPlugin> => {
  // TODO: add license check here when it's ready on the license registry
  if (strapi.features.future.isEnabled('history')) {
    return {
      register({ strapi }) {
        strapi.get('models').add(historyVersion);
      },
      bootstrap({ strapi }) {
        // Start recording history and saving history versions
        getService(strapi, 'history').init();
      },
      controllers,
      services,
      routes,
    };
  }

  /**
   * Keep registering the model to avoid losing the data if the feature is disabled,
   * or if the license expires.
   */
  return {
    register({ strapi }) {
      strapi.get('models').add(historyVersion);
    },
  };
};

export default getFeature();
