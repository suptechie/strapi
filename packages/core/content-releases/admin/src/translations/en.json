{
  "plugin.name": "Releases",
  "pages.Releases.title": "Releases",
  "pages.Releases.header-subtitle": "{number, plural, =0 {No releases} one {# release} other {# releases}}",
  "header.actions.add-release": "New Release",
  "header.actions.refresh": "Refresh",
  "header.actions.release": "Release",
  "header.actions.open-release-actions": "Release actions",
  "header.actions.edit": "Edit",
  "header.actions.delete": "Delete",
  "header.actions.created": "Created",
  "header.actions.created.description": "{number, plural, =0 {# days} one {# day} other {# days}} ago by {user}",
  "modal.release-created-notification-success": "Release created",
  "modal.add-release-title": "New Release",
  "modal.form.input.label.release-name": "Name",
  "modal.form.button.submit": "Continue",
  "pages.Details.header-subtitle": "{number, plural, =0 {No entries} one {# entry} other {# entries}}"
}
