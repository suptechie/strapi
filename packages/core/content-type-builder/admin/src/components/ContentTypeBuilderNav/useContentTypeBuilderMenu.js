import { useState } from 'react';
import { useNotification, useTracking } from '@strapi/helper-plugin';
import { camelCase, isEmpty, sortBy, toLower, upperFirst } from 'lodash';
import matchSorter from 'match-sorter';
import { useIntl } from 'react-intl';
import { useHistory } from 'react-router-dom';
import useDataManager from '../../hooks/useDataManager';
import pluginId from '../../pluginId';
import getTrad from '../../utils/getTrad';
import makeSearch from '../../utils/makeSearch';

const useContentTypeBuilderMenu = () => {
  const {
    components,
    componentsGroupedByCategory,
    contentTypes,
    isInDevelopmentMode,
    sortedContentTypesList,
  } = useDataManager();
  const toggleNotification = useNotification();
  const { trackUsage } = useTracking();
  const { formatMessage } = useIntl();
  const { push } = useHistory();
  const [search, setSearch] = useState('');

  const componentsData = sortBy(
    Object.keys(componentsGroupedByCategory).map(category => ({
      name: category,
      title: category,
      isEditable: isInDevelopmentMode,
      onClickEdit: (e, data) => {
        e.stopPropagation();

        const search = makeSearch({
          actionType: 'edit',
          modalType: 'editCategory',
          categoryName: data.name,
          header_label_1: formatMessage({
            id: getTrad('modalForm.header.categories'),
            defaultMessage: 'Categories',
          }),
          header_info_category_1: null,
          header_info_name_1: null,
          header_label_2: data.name,
          header_info_category_2: null,
          header_info_name_2: null,

          settingType: 'base',
        });

        push({ search });
      },
      links: sortBy(
        componentsGroupedByCategory[category].map(compo => ({
          name: compo.uid,
          to: `/plugins/${pluginId}/component-categories/${category}/${compo.uid}`,
          title: compo.schema.name,
        })),
        obj => obj.title
      ),
    })),
    obj => obj.title
  );

  const canOpenModalCreateCTorComponent = () => {
    return (
      !Object.keys(contentTypes).some(ct => contentTypes[ct].isTemporary === true) &&
      !Object.keys(components).some(component => components[component].isTemporary === true)
    );
  };

  const handleClickOpenModal = async (modalType, kind = '') => {
    const type = kind === 'singleType' ? kind : modalType;

    if (canOpenModalCreateCTorComponent()) {
      trackUsage(`willCreate${upperFirst(camelCase(type))}`);

      const search = makeSearch({
        modalType,
        kind,
        actionType: 'create',
        settingType: 'base',
        forTarget: modalType,
        header_label_1: 'null',
      });
      push({
        search,
      });
    } else {
      toggleNotification({
        type: 'info',
        message: {
          id: `${getTrad('notification.info.creating.notSaved')}`,
          defaultMessage:
            'Please save your work before creating a new collection type or component',
        },
      });
    }
  };

  const displayedContentTypes = sortedContentTypesList.filter(obj => obj.visible);

  const data = [
    {
      name: 'models',
      title: {
        id: `${getTrad('menu.section.models.name.')}`,
        defaultMessage: 'Collection Types',
      },
      customLink: isInDevelopmentMode && {
        id: `${getTrad('button.model.create')}`,
        defaultMessage: 'Create new collection type',
        onClick: () => handleClickOpenModal('contentType', 'collectionType'),
      },
      links: displayedContentTypes.filter(contentType => contentType.kind === 'collectionType'),
    },
    {
      name: 'singleTypes',
      title: {
        id: `${getTrad('menu.section.single-types.name.')}`,
        defaultMessage: 'Single Types',
      },
      customLink: isInDevelopmentMode && {
        id: `${getTrad('button.single-types.create')}`,
        defaultMessage: 'Create new single type',
        onClick: () => handleClickOpenModal('contentType', 'singleType'),
      },
      links: displayedContentTypes.filter(singleType => singleType.kind === 'singleType'),
    },
    {
      name: 'components',
      title: {
        id: `${getTrad('menu.section.components.name.')}`,
        defaultMessage: 'Components',
      },
      customLink: {
        id: `${getTrad('button.component.create')}`,
        defaultMessage: 'Create a new component',
        onClick: () => handleClickOpenModal('component'),
      },
      links: componentsData,
    },
  ];

  const matchByTitle = links =>
    matchSorter(links, toLower(search), { keys: [item => toLower(item.title)] });

  const getMenu = () => {
    // Maybe we can do it simpler with matchsorter wildcards ?
    return data.map(section => {
      const hasChild = section.links.some(l => !isEmpty(l.links));

      if (hasChild) {
        return {
          ...section,
          links: section.links.map(l => ({ ...l, links: matchByTitle(l.links) })),
        };
      }

      return {
        ...section,
        links: matchByTitle(section.links),
      };
    });
  };

  return { menu: getMenu(), searchValue: search, onSearchChange: setSearch };
};

export default useContentTypeBuilderMenu;
