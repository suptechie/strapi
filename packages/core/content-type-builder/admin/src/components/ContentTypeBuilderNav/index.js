import {
  SubNav,
  SubNavHeader,
  SubNavLink,
  SubNavLinkSection,
  SubNavSection,
  SubNavSections,
  Box,
} from '@strapi/parts';
import { TextButton } from '@strapi/parts/TextButton';
import { AddIcon } from '@strapi/icons';
import upperFirst from 'lodash/upperFirst';
import React from 'react';
import { useIntl } from 'react-intl';
import useContentTypeBuilderMenu from './useContentTypeBuilderMenu';
import pluginId from '../../pluginId';

// ! ASK SOUP ABOUT THE WAIT + TOGGLEPROMPT
const wait = () => new Promise(resolve => setTimeout(resolve, 100));

const ContentTypeBuilderNav = () => {
  const { menu, searchValue, onSearchChange } = useContentTypeBuilderMenu({ wait });
  const { formatMessage } = useIntl();

  return (
    <SubNav ariaLabel={`${pluginId}.plugin.name`}>
      <SubNavHeader
        searchable
        value={searchValue}
        onClear={() => onSearchChange('')}
        onChange={e => onSearchChange(e.target.value)}
        label={formatMessage({ id: `${pluginId}.plugin.name` })}
        searchLabel="Search..."
      />
      <SubNavSections>
        {menu.map(section => {
          const title = `${section.title.id}${section.links.length > 1 ? 'plural' : 'singular'}`;

          return (
            <>
              <SubNavSection
                key={section.name}
                label={formatMessage({ id: title, defaultMessage: title.defaultMessage })}
                collapsable
                badgeLabel={section.links.length.toString()}
              >
                {section.links.map(link => {
                  if (link.links) {
                    return (
                      <SubNavLinkSection key={link.title} label={upperFirst(link.title)}>
                        {link.links.map(subLink => (
                          <SubNavLink to={subLink.to} active={subLink.active} key={subLink.uid}>
                            {upperFirst(
                              formatMessage({ id: subLink.name, defaultMessage: subLink.title })
                            )}
                          </SubNavLink>
                        ))}
                      </SubNavLinkSection>
                    );
                  }

                  return (
                    <SubNavLink to={link.to} active={link.active} key={link.uid}>
                      {upperFirst(formatMessage({ id: link.name, defaultMessage: link.title }))}
                    </SubNavLink>
                  );
                })}
              </SubNavSection>
              {section.customLink && (
                <Box as="li" paddingLeft={7}>
                  <TextButton onClick={section.customLink.onClick} startIcon={<AddIcon />}>
                    {formatMessage({
                      id: section.customLink.id,
                      defaultMessage: section.customLink.id,
                    })}
                  </TextButton>
                </Box>
              )}
            </>
          );
        })}
      </SubNavSections>
    </SubNav>
  );
};

export default ContentTypeBuilderNav;
