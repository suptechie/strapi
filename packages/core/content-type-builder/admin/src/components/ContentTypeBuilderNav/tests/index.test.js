/**
 *
 * Tests for ContentTypeBuilderNav
 *
 */

import { Layout } from '@strapi/design-system';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';
import LanguageProvider from '../../../../../../admin/admin/src/components/LanguageProvider';
import Theme from '../../../../../../admin/admin/src/components/Theme';
import en from '../../../../../../admin/admin/src/translations/en.json';
import ContentTypeBuilderNav from '../index';
import mockData from './mockData';

jest.mock('../useContentTypeBuilderMenu.js', () => {
  return jest.fn(() => ({
    menu: mockData,
    searchValue: '',
    onSearchChange: () => {},
  }));
});

const makeApp = () => {
  const history = createMemoryHistory();
  const messages = { en };
  const localeNames = { en: 'English' };

  return (
    <LanguageProvider messages={messages} localeNames={localeNames}>
      <Theme>
        <Router history={history}>
          <Layout sideNav={<ContentTypeBuilderNav />}>
            <div />
          </Layout>
        </Router>
      </Theme>
    </LanguageProvider>
  );
};

describe('<ContentTypeBuilderNav />', () => {
  it('renders and matches the snapshot', () => {
    const App = makeApp(ContentTypeBuilderNav);
    const { container } = render(App);

    expect(container).toMatchSnapshot();
  });
});
