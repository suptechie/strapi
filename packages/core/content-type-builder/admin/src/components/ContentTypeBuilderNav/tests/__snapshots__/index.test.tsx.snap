// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentTypeBuilderNav /> renders and matches the snapshot 1`] = `
.c2 {
  padding-block-start: 24px;
  padding-inline-end: 16px;
  padding-block-end: 8px;
  padding-inline-start: 24px;
}

.c6 {
  background: #ffffff;
  padding-block: 8px;
  padding-inline: 8px;
  border-radius: 4px;
  border-color: #dcdce4;
  border: 1px solid #dcdce4;
  cursor: pointer;
}

.c11 {
  padding-block-start: 16px;
}

.c12 {
  background: #eaeaef;
}

.c15 {
  padding-block-start: 8px;
  padding-block-end: 16px;
}

.c18 {
  padding-block-start: 8px;
  padding-inline-end: 16px;
  padding-block-end: 8px;
  padding-inline-start: 24px;
}

.c20 {
  padding-inline-end: 24px;
  position: relative;
}

.c21 {
  text-align: left;
}

.c23 {
  padding-inline-end: 4px;
}

.c25 {
  background: #eaeaef;
  padding-inline-end: 8px;
  padding-inline-start: 8px;
  position: absolute;
  right: 0;
  top: 50%;
  min-width: 20px;
  transform: translateY(-50%);
}

.c28 {
  background: #f6f6f9;
  padding-block-start: 8px;
  padding-block-end: 8px;
  padding-inline-start: 32px;
  cursor: pointer;
}

.c31 {
  padding-inline-start: 8px;
}

.c33 {
  padding-inline-start: 32px;
}

.c34 {
  background: transparent;
  margin-block-start: 8px;
}

.c38 {
  padding-block-start: 8px;
  padding-inline-end: 16px;
  padding-block-end: 8px;
  padding-inline-start: 32px;
}

.c42 {
  background: #f6f6f9;
  padding-block-start: 8px;
  padding-block-end: 8px;
  padding-inline-start: 48px;
  cursor: pointer;
}

.c43 {
  padding-block-end: 56px;
}

.c5 {
  font-weight: 600;
  font-size: 1.8rem;
  line-height: 1.22;
  color: currentcolor;
}

.c24 {
  font-weight: 600;
  font-size: 1.1rem;
  line-height: 1.45;
  text-transform: uppercase;
  color: #666687;
}

.c32 {
  font-size: 1.4rem;
  line-height: 1.43;
  color: currentcolor;
}

.c37 {
  font-size: 1.2rem;
  line-height: 1.33;
  color: #4945ff;
}

.c41 {
  font-size: 1.4rem;
  line-height: 1.43;
  font-weight: 500;
  color: #32324d;
}

.c3 {
  align-items: flex-start;
  display: flex;
  flex-direction: column;
}

.c4 {
  align-items: flex-start;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.c7 {
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: center;
}

.c16 {
  align-items: stretch;
  display: flex;
  flex-direction: column;
  gap: 8px;
}

.c17 {
  align-items: stretch;
  display: flex;
  flex-direction: column;
  gap: 4px;
}

.c22 {
  align-items: center;
  display: flex;
  flex-direction: row;
}

.c26 {
  align-items: center;
  display: inline-flex;
  flex-direction: row;
  justify-content: center;
}

.c35 {
  align-items: center;
  display: flex;
  flex-direction: row;
  gap: 8px;
}

.c39 {
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.c36 {
  border: none;
  position: relative;
  outline: none;
}

.c36[aria-disabled='true'] {
  pointer-events: none;
}

.c36[aria-disabled='true'] svg path {
  fill: #666687;
}

.c36 svg path {
  fill: #4945ff;
}

.c36:after {
  transition-property: all;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c36:focus-visible {
  outline: none;
}

.c36:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c10 {
  border: 0;
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c27 {
  border-radius: 4px;
  height: 2.4rem;
}

.c8 {
  position: relative;
  outline: none;
}

.c8[aria-disabled='true'] {
  pointer-events: none;
}

.c8:after {
  transition-property: all;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c8:focus-visible {
  outline: none;
}

.c8:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c13 {
  height: 1px;
  border: none;
  flex-shrink: 0;
  margin: 0;
}

.c9 {
  border-color: #dcdce4;
  height: 3.2rem;
  width: 3.2rem;
  color: #8e8ea9;
}

.c9:hover,
.c9:focus {
  color: #666687;
}

.c9[aria-disabled='true'] {
  color: #666687;
}

.c1 {
  width: 23.2rem;
  background: #f6f6f9;
  position: sticky;
  top: 0;
  height: 100vh;
  overflow-y: auto;
  border-right: 1px solid #dcdce4;
  z-index: 1;
}

.c14 {
  width: 2.4rem;
  background-color: #dcdce4;
}

.c29 {
  display: flex;
  align-items: center;
  justify-content: space-between;
  text-decoration: none;
  color: #32324d;
}

.c29 svg>* {
  fill: #666687;
}

.c29.active {
  background-color: #f0f0ff;
  border-right: 2px solid #4945ff;
  color: #271fe0;
  font-weight: 500;
}

.c29:focus-visible {
  outline-offset: -2px;
}

.c30 {
  width: 0.4rem;
  height: 0.4rem;
  background-color: #666687;
  border-radius: 50%;
}

.c40 {
  border: none;
  padding: 0;
  background: transparent;
  display: flex;
  align-items: center;
}

.c19>svg {
  height: 0.4rem;
  fill: #8e8ea9;
}

.c0 {
  display: grid;
  grid-template-columns: auto 1fr;
}

.c44 {
  overflow-x: hidden;
}

<div>
  <div
    class="c0"
  >
    <nav
      aria-label="Content-Types Builder"
      class="c1"
    >
      <div
        class="c2 c3"
      >
        <div
          class="c4"
        >
          <h2
            class="c5"
          >
            Content-Types Builder
          </h2>
          <span>
            <button
              aria-disabled="false"
              aria-labelledby=":r1:"
              class="c6 c7 c8 c9"
              tabindex="0"
              type="button"
            >
              <span
                class="c10"
              >
                Search
              </span>
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="16"
                viewBox="0 0 32 32"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M29.061 26.939 23.125 21A11.515 11.515 0 1 0 21 23.125l5.941 5.942a1.503 1.503 0 0 0 2.125-2.125zM5.5 14a8.5 8.5 0 1 1 8.5 8.5A8.51 8.51 0 0 1 5.5 14"
                />
              </svg>
            </button>
          </span>
        </div>
        <div
          class="c11"
        >
          <hr
            class="c12 c13 c14"
          />
        </div>
      </div>
      <div
        class="c15"
      >
        <ol
          class="c16"
        >
          <li>
            <div
              class="c17"
            >
              <div
                class="c18 c19"
              >
                <div
                  class="c20"
                >
                  <button
                    aria-controls=":r3:"
                    aria-expanded="true"
                    class="c21 c22"
                  >
                    <div
                      class="c23"
                    >
                      <span
                        class="c24"
                      >
                        Collection Types
                      </span>
                    </div>
                    <svg
                      aria-hidden="true"
                      fill="#8e8ea9"
                      height="16"
                      style="transform: rotateX(0deg);"
                      viewBox="0 0 32 32"
                      width="1.2rem"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m26.708 12.708-10 10a1 1 0 0 1-1.415 0l-10-10A1 1 0 0 1 6 11h20a1 1 0 0 1 .707 1.707"
                      />
                    </svg>
                  </button>
                  <div
                    class="c25 c26 c27"
                  >
                    <span
                      class="c24"
                    >
                      2
                    </span>
                  </div>
                </div>
              </div>
              <ol
                id=":r3:"
              >
                <li>
                  <a
                    aria-disabled="false"
                    class="c28 c29"
                    href="/plugins/content-type-builder/content-types/application::address.address"
                    target="_self"
                  >
                    <div
                      class="c22"
                    >
                      <span
                        class="c30"
                      />
                      <span
                        class="c31 c32"
                      >
                        Address
                      </span>
                    </div>
                  </a>
                </li>
                <li>
                  <a
                    aria-disabled="false"
                    class="c28 c29"
                    href="/plugins/content-type-builder/content-types/application::category.category"
                    target="_self"
                  >
                    <div
                      class="c22"
                    >
                      <span
                        class="c30"
                      />
                      <span
                        class="c31 c32"
                      >
                        Category
                      </span>
                    </div>
                  </a>
                </li>
              </ol>
            </div>
            <div
              class="c33"
            >
              <button
                aria-disabled="false"
                class="c34 c35 c36"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="0.8rem"
                  viewBox="0 0 32 32"
                  width="0.8rem"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M28.5 16a1.5 1.5 0 0 1-1.5 1.5h-9.5V27a1.5 1.5 0 1 1-3 0v-9.5H5a1.5 1.5 0 1 1 0-3h9.5V5a1.5 1.5 0 1 1 3 0v9.5H27a1.5 1.5 0 0 1 1.5 1.5"
                  />
                </svg>
                <span
                  class="c37"
                >
                  Create new collection type
                </span>
              </button>
            </div>
          </li>
          <li>
            <div
              class="c17"
            >
              <div
                class="c18 c19"
              >
                <div
                  class="c20"
                >
                  <button
                    aria-controls=":r4:"
                    aria-expanded="true"
                    class="c21 c22"
                  >
                    <div
                      class="c23"
                    >
                      <span
                        class="c24"
                      >
                        Single Types
                      </span>
                    </div>
                    <svg
                      aria-hidden="true"
                      fill="#8e8ea9"
                      height="16"
                      style="transform: rotateX(0deg);"
                      viewBox="0 0 32 32"
                      width="1.2rem"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m26.708 12.708-10 10a1 1 0 0 1-1.415 0l-10-10A1 1 0 0 1 6 11h20a1 1 0 0 1 .707 1.707"
                      />
                    </svg>
                  </button>
                  <div
                    class="c25 c26 c27"
                  >
                    <span
                      class="c24"
                    >
                      1
                    </span>
                  </div>
                </div>
              </div>
              <ol
                id=":r4:"
              >
                <li>
                  <a
                    aria-disabled="false"
                    class="c28 c29"
                    href="/plugins/content-type-builder/content-types/application::homepage.homepage"
                    target="_self"
                  >
                    <div
                      class="c22"
                    >
                      <span
                        class="c30"
                      />
                      <span
                        class="c31 c32"
                      >
                        Homepage
                      </span>
                    </div>
                  </a>
                </li>
              </ol>
            </div>
            <div
              class="c33"
            >
              <button
                aria-disabled="false"
                class="c34 c35 c36"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="0.8rem"
                  viewBox="0 0 32 32"
                  width="0.8rem"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M28.5 16a1.5 1.5 0 0 1-1.5 1.5h-9.5V27a1.5 1.5 0 1 1-3 0v-9.5H5a1.5 1.5 0 1 1 0-3h9.5V5a1.5 1.5 0 1 1 3 0v9.5H27a1.5 1.5 0 0 1 1.5 1.5"
                  />
                </svg>
                <span
                  class="c37"
                >
                  Create new single type
                </span>
              </button>
            </div>
          </li>
          <li>
            <div
              class="c17"
            >
              <div
                class="c18 c19"
              >
                <div
                  class="c20"
                >
                  <button
                    aria-controls=":r5:"
                    aria-expanded="true"
                    class="c21 c22"
                  >
                    <div
                      class="c23"
                    >
                      <span
                        class="c24"
                      >
                        Components
                      </span>
                    </div>
                    <svg
                      aria-hidden="true"
                      fill="#8e8ea9"
                      height="16"
                      style="transform: rotateX(0deg);"
                      viewBox="0 0 32 32"
                      width="1.2rem"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m26.708 12.708-10 10a1 1 0 0 1-1.415 0l-10-10A1 1 0 0 1 6 11h20a1 1 0 0 1 .707 1.707"
                      />
                    </svg>
                  </button>
                  <div
                    class="c25 c26 c27"
                  >
                    <span
                      class="c24"
                    >
                      2
                    </span>
                  </div>
                </div>
              </div>
              <ol
                id=":r5:"
              >
                <li>
                  <div
                    class=""
                  >
                    <div
                      class="c38"
                    >
                      <div
                        class="c39"
                      >
                        <button
                          aria-controls=":r6:"
                          aria-expanded="true"
                          class="c40"
                        >
                          <svg
                            aria-hidden="true"
                            fill="#4a4a6a"
                            height="1.2rem"
                            style="transform: rotateX(0deg);"
                            viewBox="0 0 32 32"
                            width="1.2rem"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="m26.708 12.708-10 10a1 1 0 0 1-1.415 0l-10-10A1 1 0 0 1 6 11h20a1 1 0 0 1 .707 1.707"
                            />
                          </svg>
                          <div
                            class="c31"
                          >
                            <span
                              class="c41"
                            >
                              Basic
                            </span>
                          </div>
                        </button>
                      </div>
                    </div>
                    <ul
                      id=":r6:"
                    >
                      <li>
                        <a
                          aria-disabled="false"
                          class="c42 c29"
                          href="/plugins/content-type-builder/component-categories/basic/basic.simple"
                          target="_self"
                        >
                          <div
                            class="c22"
                          >
                            <span
                              class="c30"
                            />
                            <span
                              class="c31 c32"
                            >
                              Simple
                            </span>
                          </div>
                        </a>
                      </li>
                    </ul>
                  </div>
                </li>
                <li>
                  <div
                    class=""
                  >
                    <div
                      class="c38"
                    >
                      <div
                        class="c39"
                      >
                        <button
                          aria-controls=":r7:"
                          aria-expanded="true"
                          class="c40"
                        >
                          <svg
                            aria-hidden="true"
                            fill="#4a4a6a"
                            height="1.2rem"
                            style="transform: rotateX(0deg);"
                            viewBox="0 0 32 32"
                            width="1.2rem"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="m26.708 12.708-10 10a1 1 0 0 1-1.415 0l-10-10A1 1 0 0 1 6 11h20a1 1 0 0 1 .707 1.707"
                            />
                          </svg>
                          <div
                            class="c31"
                          >
                            <span
                              class="c41"
                            >
                              Default
                            </span>
                          </div>
                        </button>
                      </div>
                    </div>
                    <ul
                      id=":r7:"
                    >
                      <li>
                        <a
                          aria-disabled="false"
                          class="c42 c29"
                          href="/plugins/content-type-builder/component-categories/default/default.closingperiod"
                          target="_self"
                        >
                          <div
                            class="c22"
                          >
                            <span
                              class="c30"
                            />
                            <span
                              class="c31 c32"
                            >
                              Closingperiod
                            </span>
                          </div>
                        </a>
                      </li>
                      <li>
                        <a
                          aria-disabled="false"
                          class="c42 c29"
                          href="/plugins/content-type-builder/component-categories/default/default.dish"
                          target="_self"
                        >
                          <div
                            class="c22"
                          >
                            <span
                              class="c30"
                            />
                            <span
                              class="c31 c32"
                            >
                              Dish
                            </span>
                          </div>
                        </a>
                      </li>
                    </ul>
                  </div>
                </li>
              </ol>
            </div>
            <div
              class="c33"
            >
              <button
                aria-disabled="false"
                class="c34 c35 c36"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="0.8rem"
                  viewBox="0 0 32 32"
                  width="0.8rem"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M28.5 16a1.5 1.5 0 0 1-1.5 1.5h-9.5V27a1.5 1.5 0 1 1-3 0v-9.5H5a1.5 1.5 0 1 1 0-3h9.5V5a1.5 1.5 0 1 1 3 0v9.5H27a1.5 1.5 0 0 1 1.5 1.5"
                  />
                </svg>
                <span
                  class="c37"
                >
                  Create a new component
                </span>
              </button>
            </div>
          </li>
        </ol>
      </div>
    </nav>
    <div
      class="c43 c44"
    >
      <div />
    </div>
  </div>
  <span
    class="c10"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </span>
</div>
`;
